client/my-sites/site/index.jsx

```diff
@@ -16,7 +16,24 @@ module.exports = React.createClass( {
 	displayName: 'Site',
 
 	componentDidMount: function() {
-		debug( 'The Site component is mounted.' );
+		if ( this.props.isSelected ) {
+			this.scrollIntoView();
+		}
+	},
+
+	componentDidUpdate: function( prevProps, prevState ) {
+		if ( this.props.isSelected && ! prevProps.isSelected ) {
+			this.scrollIntoView();
+		}
+	},
+
+	scrollIntoView: function() {
+		var node = ReactDom.findDOMNode( this ),
```

Comment:
Rather than pulling in `react-dom` to use the `findDOMNode` method, you could alternatively use a `ref` on the rendered root `div`, which would make this method a bit simpler and lose the extra dependency:

```jsx
scrollIntoView() {
	const parentScrollTop = this.refs.site.offsetTop;

	if ( this.refs.site.previousSibling ) {
		parentScrollTop -= this.refs.site.previousSibling.offsetHeight / 2;
	}
	
	this.refs.site.parentNode.scrollTop = parentScrollTop;
},

render() {
	return <div ref="site" className={ siteClass } />;
}
```